### PR TITLE
Update mypy

### PR DESCRIPTION
**Pull request by gsnedders**
Supplants #39101 with all the necessary changes (a lot of this is just removing redundant casts and ignores, but also making sure mypy only ends up scanning wptrunner/wptserve/etc. once each rather than as both e.g.`tools.wptrunner.wptrunner` and `wptrunner`).

### CODE REVIEW

**Review by bashi — APPROVED** [review]
tools/webrtransport lgtm.

**Review by nidhijaju — APPROVED** [review]
webtransport lgtm

File: tools/webdriver/webdriver/client.py [hunk 1]
```diff
@@ -313,24 +313,24 @@ def close(self):
     # The many "type: ignore" comments here and below are to silence mypy's
```
Comment (jgraham) [inline tools/webdriver/webdriver/client.py:313]: Seems like this comment should also go?

**Review by jgraham — APPROVED** [review]
r+ but please remove the leftover comment.

**Pull request merged**